public/app/plugins/datasource/prometheus/module.ts

```diff
@@ -1,11 +1,27 @@
 import { DataSourcePlugin } from '@grafana/data';
+import {
+  PrometheusDatasource as PrometheusDatasourcePackage,
+  PromQueryEditorByApp as PromQueryEditorByAppPackage,
+  PromCheatSheet as PromCheatSheetPackage,
+} from '@grafana/prometheus';
+import { config } from '@grafana/runtime';
 
 import PromCheatSheet from './components/PromCheatSheet';
 import PromQueryEditorByApp from './components/PromQueryEditorByApp';
 import { ConfigEditor } from './configuration/ConfigEditor';
+import { ConfigEditor as ConfigEditorPackage } from './configuration/ConfigEditorPackage';
 import { PrometheusDatasource } from './datasource';
 
-export const plugin = new DataSourcePlugin(PrometheusDatasource)
-  .setQueryEditor(PromQueryEditorByApp)
-  .setConfigEditor(ConfigEditor)
-  .setQueryEditorHelp(PromCheatSheet);
+const usePackage = config.featureToggles.usePrometheusFrontendPackage;
+
+const PrometheusDataSourceUsed = usePackage ? PrometheusDatasource : PrometheusDatasourcePackage;
+const PromQueryEditorByAppUsed = usePackage ? PromQueryEditorByApp : PromQueryEditorByAppPackage;
+const ConfigEditorUsed = usePackage ? ConfigEditor : ConfigEditorPackage;
+const PromCheatSheetUsed = usePackage ? PromCheatSheet : PromCheatSheetPackage;
+
+// @ts-ignore
+export const plugin = new DataSourcePlugin(PrometheusDataSourceUsed)
+  // @ts-ignore
```

Comment:
Could you please add a comment here like we are gonna remove those `@ts-ignore` lines when we'll fully on prom-package. I know you can write something better :)

Reply:
Yes! I will continue debugging the reason for these type errors and add a comment